lib/Dialect/LWE/IR/LWEDialect.cpp

```diff
@@ -177,6 +178,30 @@ LogicalResult ReinterpretUnderlyingTypeOp::verify() {
   return success();
 }
 
+// Verification for RLWE_EncryptOp
+LogicalResult RLWEEncryptOp::verify() {
+  Type keyType = getKey().getType();
+  lwe::RLWEParamsAttr keyParams =
+      llvm::TypeSwitch<Type, lwe::RLWEParamsAttr>(keyType)
+          .Case<lwe::RLWEPublicKeyType, lwe::RLWESecretKeyType>(
+              [](auto key) { return key.getRlweParams(); })
+          .Default([](Type) {
+            llvm_unreachable("impossible by type constraints");
+            return nullptr;
+          });
+
+  if (getOutput().getType().getRlweParams() != keyParams) {
```

Comment:
note to self: this isn't implemented as `RlweParametersMatch` trait because it uses `RLWESecretOrPublicKey`